etp-backend/src/test/clj/solita/etp/service/valvonta_kaytto/toimenpide_test.clj

```diff
@@ -60,3 +60,18 @@
     (t/is (true? (toimenpide/sakkopaatos-toimenpide? {:type-id 14}))))
   (t/testing "Käskypäätös / kuulemiskirje is not recognized as sakkopäätös-toimenpide?"
     (t/is (false? (toimenpide/sakkopaatos-toimenpide? {:type-id 7})))))
+
+(t/deftest sakkopaatos-tiedoksianto-toinen-postitus-test
+  (let [
+        sakkopaatos-toinen-postitus-type-id 17
+        sakkopaatos-toinen-postitus-type-key :penalty-decision-notice-second-mailing
+        ]
+    (t/testing "Sakkopäätös / tiedoksianto (toinen postitus) has type-key 17"
+      (t/is (= (toimenpide/type-key sakkopaatos-toinen-postitus-type-id)
+               sakkopaatos-toinen-postitus-type-key)))
+    (t/testing "Sakkopäätös / tiedoksianto (toinen postitus) is recognized as sakkopaatos-toimenpide?"
+      (t/is (true? (toimenpide/sakkopaatos-toimenpide? {:type-id sakkopaatos-toinen-postitus-type-id}))))
+    (t/testing "Sakkopäätös / tiedoksianto (toinen postitus) is not recognized as asha-toimenpide?"
+      (t/is (false? (toimenpide/asha-toimenpide? {:type-id sakkopaatos-toinen-postitus-type-id}))))))
+
```

Comment:
Kirjoittelin yksikkötestit tänne. Onkohan tämä hyvä? Mietin, että tuohon let blockkiin olisi ollut kiva kirjoittaa
```clojure
   predicates-and-their-expected-values
        {toimenpide/asha-toimenpide?        false
         toimenpide/sakkopaatos-toimenpide? true
         }
```
Mutta ei nyt ihan onnistunut..

Reply:
Ai näin?
```
  (let [sakkopaatos-toinen-postitus-type-id 17
        sakkopaatos-toinen-postitus-type-key :penalty-decision-notice-second-mailing
        predicates-and-their-expected-values {toimenpide/asha-toimenpide?        false
                                              toimenpide/sakkopaatos-toimenpide? true}]
    (doseq [[pred expected-value] predicates-and-their-expected-values]
      (t/is (= expected-value (pred {:type-id 17}))))

```
Voisi toimia, jos tuohon lisäisi selitetekstit, muutoin ehkä ärsyttävä debugata mikä ei mennyt läpi.

Lisäksi tuo voisi varmaan olla sisäkkäisiä listoja, eikä map

Reply:
Laittaisin tässä nuo arvot ihan vain suoraan käyttöpaikkoihin enkä lettiin mitään. Yksittäisen testin lukeminen helpottuu, kun kaiken näkee yhdellä vilkaisulla. Nyt silmät joutuu hyppimään letin ja letattujen asioiden käyttöpaikkojen välilä. Kyse on kuitenkin vain yksittäisistä arvoista, joten toistosta ei juuri ole haittaa.

Reply:
Tähän näytti olevan myös makro
```
    (t/are [pred expected-value] (= expected-value (pred {:type-id 17}))
                                 toimenpide/asha-toimenpide? false
                                 toimenpide/sakkopaatos-toimenpide? false)
```

Reply:
Sinänsä joku tänkaltainen vähän datalähtöisempi testaus vois olla minusta hyvä. Tässä keississä en oo ihan varma kun funktiokin on osa dataa, mikä tekee tästä vähän sekavamman luettavan, mutta se voi vaan olla minun epäfunktionaaliset aivot

Reply:
Tuon are-makron raportointi toimi ihan hyvin, eli sen kans ei tarvis mitään selitetekstejäkään